File: ShadowFrameworkJS/src/shadow/geometry/data/SFFixedFloat16.js

var SFFixedFloat16_BIT_SIZE=16;
var SFFixedFloat16_MULTIPLIER=0.001;
var SFFixedFloat16_BACKMULTIPLIER=1000;

function SFFixedFloat16(f){
	this.multiplier=SFFixedFloat16_MULTIPLIER;
	this.backmultiplier=SFFixedFloat16_BACKMULTIPLIER;
	if(!(f==undefined)){
		this.setFloat(f);
	}
}

inherit(SFFixedFloat16,SFGenericFixedFloat);


SFFixedFloat16.prototype["getFloat"]=function(){
			var value=this.value>=32768?(this.value-65536):this.value;
			return value*this.multiplier;
		};

SFFixedFloat16.prototype["clone"]=function(){
	return new SFFixedFloat16();
};

SFFixedFloat16.prototype["getBitSize"]=function(){
	return SFFixedFloat16_BIT_SIZE;
};

File: trunk/sfoglintegration/SF20Data/src/shadow/system/data/formats/SFFixedFloat16.java
package shadow.system.data.formats;

import shadow.system.data.objects.SFGenericFixedFloat;


public class SFFixedFloat16 extends SFGenericFixedFloat{

	private static int BIT_SIZE=16;
	static float MULTIPLIER=0.001f;
	static float BACKMULTIPLIER=1000f;
	
	public SFFixedFloat16() {
		super(MULTIPLIER,BACKMULTIPLIER);
	}
	
	public SFFixedFloat16(float multiplier, float backmultiplier) {
		super(multiplier,backmultiplier);
	}

	public SFFixedFloat16(float f) {
		super(MULTIPLIER,BACKMULTIPLIER);
		setFloat(f);
	}
	
	@Override
	public SFFixedFloat16 copyDataObject() {
		return new SFFixedFloat16();
	}
	
	@Override
	public int getBitSize() {
		return BIT_SIZE;
	}
}
